src/Raven.NewClient/Blittable/EntityToBlittable.cs

```diff
@@ -79,7 +80,7 @@ public object ConvertToEntity(Type entityType, string id, BlittableJsonReaderObj
                 if (documentType != null)
                 {
                     var type = Type.GetType(documentType);
-                    if (type != null)   
+                    if (type != null && entityType.IsAssignableFrom(type))
```

Comment:
It's a feature :-)
The idea is that it is very common to get complaints about it. but this avoid the problem